docs/contributing/code-contributions.md

```diff
@@ -0,0 +1,56 @@
+
+This page includes instructions specific to the Gatsby core codebase, blog, and website. To start setting up the Gatsby repo on your machine, check out the page on [setting up your local dev environment](/contributing/setting-up-your-local-dev-environment/). For additional instructions on contributing to the docs, visit the [docs contributions page](/contributing/docs-contributions/).
+
+## Making changes to the website
+
+If you want to make more changes to the website functionality or documentation, that is, change layout components or templates, add sections/pages, follow the steps for [setting up your local dev environment](/contributing/setting-up-your-local-dev-environment/). You can then spin up your own instance of the Gatsby website and make/preview your changes before raising
+a pull request.
+
+## Making changes to the starter library
+
+Note: You don't need to follow these steps to submit to the starter library. This is only necessary if you'd like to contribute to the functionality of the starter library. To submit a starter, [follow these steps instead](/contributing/submit-to-starter-library/).
+
+To develop on the starter library, you'll need to supply a GitHub personal access token.
+
+1. Create a personal access token in your GitHub [Developer settings](https://github.com/settings/tokens).
+2. In the new token's settings, grant that token the "public_repo" scope.
+3. Create a file in the root of `www` called `.env.development`, and add the token to that file like so:
+
+```text:title=.env.development
+GITHUB_API_TOKEN=YOUR_TOKEN_HERE
+```
+
+The `.env.development` file is ignored by git. Your token should never be committed.
+
+## Contributing to the blog
+
+Note: Before adding a blog post ensure you have approval from a member of the Gatsby team. You can do this by [opening an issue](https://github.com/gatsbyjs/gatsby/issues/new/choose) or contacting [@gatsbyjs on Twitter](https://twitter.com/gatsbyjs).
+
+To add a new blog post to the gatsbyjs.org blog:
+
+- Clone [the Gatsby repo](https://github.com/gatsbyjs/gatsby/) and navigate to `/www`
+- Run `yarn` to install all of the website's dependencies.
+- Run `npm run develop` to preview the blog at `http://localhost:8000/blog`.
+- The content for the blog lives in the `/docs/blog` folder. Make additions or modifications here.
+- Add your avatar image to `/docs/blog/avatars`
+- Add your name to `/docs/blog/author.yaml`
+- Add a new folder following the pattern `/docs/blog/yyyy-mm-dd-title`. Within this newly created folder add an `index.md` file.
+- Add `title`, `date`, `author`, and `tags` ([view existing tags](https://www.gatsbyjs.org/blog/tags/) or add a new one) to the frontmatter of your `index.md`. If you are cross posting your post you can add `canonicalLink` for SEO benefits. You can check the other blog posts in `/docs/blog` for examples.
```

Comment:
Would it be valuable to have an example here? I've heard that the general term "frontmatter" isn't always known to people:

````
```md
---
title: "Your Great Blog Post"
date: YYYY-MM-DD
author: John Doe
tags:
  - awesome
  - post
---

Your next great blog post awaits.
```
````

Reply:
Great idea. I seem to remember it only allows a singular author, is that still true? Are tags optional?

Reply:
> I seem to remember it only allows a singular author, is that still true

Correct! But it would be fairly trivial to get multiple author support if we want that

> Are tags optional

Yep! If a post doesn't have tags, we just render `null`. It's probably a best practice to add at least one though.